src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.cs

```diff
@@ -38,6 +38,13 @@ public static SslStreamCertificateContext Create(X509Certificate2 target, X509Ce
                     NetEventSource.Error(null, $"Failed to build chain for {target.Subject}");
                 }
 
+                if (chain.ChainElements.Count == 0)
```

Comment:
Should the Android X509Chain.Build() return at least the certificate it self? That how it works on other platforms when building complete chain works AFAIK. (@bartonjs would know for sure)

We could also wrap the chain processing in the `if count > 0`as `intermediates` is already initialized to `Array.Empty<X509Certificate2>` and we can fall-through and have singe exit.

I'm generally fine with the change but I worry that this may bit us (or other users) elsewhere.

Reply:
> return at least the certificate it self

We could certainly do that (and it sounds reasonable to my unlearned ears). Thoughts, @bartonjs?

Reply:
It was a thing I thought about suggesting, but it probably requires doing some unnatural things.  For instance, the Android chain will fail to populate if the cert is expired... so if we make the single cert show up it needs both PartialChain and NotTimeValid.  Now repeat that for all of the other codes that make the Android chain fail and we're talking a lot of custom code to move chain.Certificates.Count from 0 to 1.

If cost were no factor, it's the better state to be in.  If it makes something here massively simpler then maybe it's worth it.  Otherwise, the "lots of things make this chain come back empty" state is... obviously different enough that when it matters you kind of need to know about it.

Reply:
Okay, I have left this as dealing with a potentially zero-element chain, but switched to putting the processing in `if` and falling through to the single exit.